projects/npm-tools/packages/npm-scripts/src/scripts/build.js

```diff
@@ -134,15 +149,75 @@ module.exports = async function (...args) {
 		}
 
 		if (Array.isArray(BUILD_CONFIG.exports)) {
-			await runWebpackAsBundler(BUILD_CONFIG);
+			fs.mkdirSync(BUILD_CONFIG.output, {recursive: true});
+
+			let pkgJson = require(path.resolve('package.json'));
+
+			if (!BUILD_CONFIG.bundler) {
+				pkgJson = {
+					...pkgJson,
+				};
 
-			createEsm2AmdExportsBridges(
+				pkgJson.main = 'index.js';
+				delete pkgJson.dependencies;
+				delete pkgJson.devDependencies;
+
+				fs.writeFileSync(
+					path.join(BUILD_CONFIG.output, 'package.json'),
+					JSON.stringify(pkgJson, null, '\t'),
+					'utf8'
+				);
+			}
+
+			await runWebpackAsBundler(
 				CWD,
-				BUILD_CONFIG.output,
-				BUILD_CONFIG.exports
+				BUILD_CONFIG,
+				getMergedConfig('babel')
+			);
+
+			let manifest;
+
+			try {
+				manifest = require(path.resolve(
+					BUILD_CONFIG.output,
+					'manifest.json'
+				));
+			}
+			catch (error) {
+				manifest = {
+					packages: {
+						'/': {
+							dest: {
+								dir: '.',
+								id: '/',
+								name: pkgJson.name,
+								version: pkgJson.version,
+							},
+							modules: {},
+							src: {
+								id: '/',
+								name: pkgJson.name,
+								version: pkgJson.version,
+							},
+						},
+					},
+				};
+			}
+
+			createEsm2AmdIndexBridge(CWD, BUILD_CONFIG, manifest);
+
+			createEsm2AmdExportsBridges(CWD, BUILD_CONFIG, manifest);
+
+			fs.writeFileSync(
+				path.join(BUILD_CONFIG.output, 'manifest.json'),
+				JSON.stringify(manifest, null, '\t'),
+				'utf8'
 			);
 		}
 		else if (BUILD_CONFIG.exports) {
+
+			// TODO: remove this
```

Comment:
nit: Maybe we can add a bit more of a description here so that we don't get confused later.

Reply:
Well, my plan is to do a full rewrite of the commits and a refactor once everything is working, because everything is a bit suboptimal as it is now... :sweat_smile:

Reply:
I've enhanced the comment in a new push force.